### ISSUE TITLE
ENH: Initilize Print Options from Environment Variables

### ISSUE THREAD

**Issue by daskol** [numpy/numpy#22188]
### Proposed new feature or change:

At the moment, print options are initialized with default options and can be changed with either `numpy.set_printoptions` function or `numpy.printoptions` context manager. Sometimes it is not very convenient to modify a code with mentioned above function to display arrays in more readable way.

It would be great to initialize print options from environment variables prefixed with `NUMPY_PRINTOPTION_` (e.g. `precision` print option should be initialized from `NUMPY_PRINTOPTION_PRECISION` envvar).
```shell
NUMPY_PRINTOPTION_LINEWIDTH=120 python my-fance-solver.py
```

**Comment by LeonaTaric** [comment]
good idea

**Comment by daskol** [comment]
> good idea
Thanks. I am still waiting for triage.

**Comment by mattip** [comment]
Environment variables seems like a bad fit for this. Perhaps a different mechanism to parse global state options would be better: some kind of config file [like matplotlib](https://matplotlib.org/stable/tutorials/introductory/customizing.html#the-matplotlibrc-file) has?
> Sometimes it is not very convenient to modify a code with mentioned above function to display arrays in more readable way.

Could you describe your use case a bit more? Is it really so hard to add a snippet to set the `printoptions` at startup?

**Comment by daskol** [comment]
> Environment variables seems like a bad fit for this.

Why? For example, `make` verbosity is controlled with `VERBOSE=1` envvar or Google's `glog` logger can be fine-tuned with `GLOG_*` variables. By the way, matplotlib works with envvar as well.

> Could you describe your use case a bit more?

From my perspective, there are several aspects of the issue.

The first one is that line width equals 80 is a good choice in general but nowadays we have wider screens and terminals and this value should be adjusted somehow in order to print more information and use benefits of wide displays. In this case, `.mpstyle`-like files can be a good solution to configure basic settings. However, NumPy print options are not dynamically adjusted in case if someone resizes the terminal (e.g. GUI or TMUX).

Another one aspect is that use of `np.printoptions` requires source code modifications. It requires text editor which is not always available (e.g. isolated environment  like docker container may not have an editor). Also, editing text is plain and simple but not simpler than just prepending with envvar on demand as follows.

```shell
NUMPY_PRINTOPTION_LINEWIDTH=160 python run-my-solver.py --opt1 --flag2 posisional3
```

In my opinion, envvar is a good well-known widely-used non-invasive and easy way to adjust configuration in general and printing options in particular.

**Comment by rossbar** [comment]
I agree with @mattip and am leery about environment variables, as this approach would then be subject to platform dependencies.

The next step here would be to post this proposal to the [NumPy mailing list](https://mail.python.org/mailman3/lists/numpy-discussion.python.org/), as that's the correct forum for proposing new features.

**Comment by daskol** [comment]
>  this approach would then be subject to platform dependencies.

And this is okay. Envvars exist on the majority of platforms which are Linux, BSD, macOS, Windows, and even DOS. I am pretty sure that the majority of developers works on these platforms and I am not sure that Python can even be built on platforms which are not in the list (e.g. WebAssembly). The latter means that compatibility issues are reduced to existence of a special function (namely, `os.getenv`). In other words, Python is a compatibility layer itself.

Thank you, I'll send the proposal to the mailing list.
